pkg/runtime-manager/resource-executor/resource_executor.go

```diff
@@ -0,0 +1,38 @@
+/*
+
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource_executor
+
+type RuntimeResourceExecutor interface {
+	GetMetaInfo() string
+	GenerateResourceCheckpoint() interface{}
+	GenerateHookRequest() interface{}
+	ParseRequest(request interface{}) error
+	ResourceCheckPoint(response interface{}) error
+}
+
+type RuntimeResourceType string
+
+const (
+	RuntimePodResource       RuntimeResourceType = "RuntimePodResource"
```

Comment:
> > I think we can merge the pod resource and container resource, because a container must belong to a pod.
> > the meta info is same for docker and cri, we need a unified struct for docker and cri
> 
> because in StartCcontainer/UpdateContainer api, there is only containerID, so we should have a db indexed by containerID, otherwise we may loop the whole db to find specify container info.

the repo stores unified struct of container/pod for both containerd/dockerd sceario. the docker resource-executor should convert the docker request to the unified struct and store